packages/x-charts/src/ResponsiveChartContainer/index.tsx

```diff
@@ -1,53 +1,120 @@
 import * as React from 'react';
-import { ResizeObserver } from '@juggle/resize-observer';
+import useEnhancedEffect from '@mui/utils/useEnhancedEffect';
+import ownerWindow from '@mui/utils/ownerWindow';
+import { styled } from '@mui/material/styles';
 import { ChartContainer, ChartContainerProps } from '../ChartContainer';
 import { MakeOptional } from '../models/helpers';
 
 const useChartDimensions = (
   inWidth?: number,
   inHeight?: number,
 ): [React.RefObject<HTMLDivElement>, number, number] => {
-  const ref = React.useRef<HTMLDivElement>(null);
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const displayError = React.useRef<boolean>(false);
 
   const [width, setWidth] = React.useState(0);
   const [height, setHeight] = React.useState(0);
 
+  const computeSizeAndPublishResizeEvent = React.useCallback(() => {
```

Comment:
Am I missing something or is the function name somewhat misleading?
I don't see any `publishing` of a `resizeEvent`... 🙈

Reply:
It's the grid name which is coherent, but I agree here it does not make sense

Reply:
Effectively, I was too focused on the content I forget the naming 🙈